Client/Frontend/Browser/Favicons/UIImageView+Favicon.swift

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import UIKit
+import BraveShared
+import BraveFavicon
+
+extension UIImageView {
+
+  private struct AssociatedObjectKeys {
+    static var faviconTask: Int = 0
+  }
+
+  /// The associated favicon fetcher for a UIImageView to ensure we don't
+  /// immediately cancel the FaviconFetcher load when the function call goes
+  /// out of scope.
+  ///
+  /// Must use objc associated objects because we are extending UIKit in an
+  /// extension
+  private var faviconTask: Task<Void, Error>? {
+    get { objc_getAssociatedObject(self, &AssociatedObjectKeys.faviconTask) as? Task<Void, Error> }
+    set { objc_setAssociatedObject(self, &AssociatedObjectKeys.faviconTask, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
+  }
+
+  /// Load the favicon from a site URL directly into a `UIImageView`. If no
+  /// favicon is found, a monogram will be used where the letter is determined
+  /// based on the site URL.
+  func loadFavicon(for siteURL: URL, monogramFallbackCharacter: Character? = nil) {
+    cancelFaviconLoad()
+    faviconTask = Task { @MainActor in
+      let favicon = try? await FaviconFetcher.loadIcon(url: siteURL, persistent: !PrivateBrowsingManager.shared.isPrivateBrowsing)
+      self.image = favicon?.image
+    }
+  }
+
+  /// Cancel any pending favicon load task. This is to prevent race condition UI glitches with reusable table/collection view cells.
+  func cancelFaviconLoad() {
+    faviconTask?.cancel()
+    faviconTask = nil
+  }
+  
+  func clearMonogramFavicon() {
+    
```

Comment:
I believe we should clear the monogram here :)

Reply:
Nothing to remove here. The monogram is part of the image itself instead of a UILabel. Still trying to figure out whether we should change it to a label.

Reply:
Can we have comment here why it empty here in that case? It has been used in many places and it can be if someone came upon it and wonder why it is empty method.